portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,35 +26,63 @@
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.sps.data.Comment;
 
-
-/** Servlet that returns some example content. */
+/** Servlet that returns data for my portfolio. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private List comments = new ArrayList<String>();
+  private List comments = new ArrayList<Comment>();
+  private static final int DEFAULT_NUM_COMMENTS = 10;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Load from datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    int commentsMaxNum = DEFAULT_NUM_COMMENTS;
+    try {
+      commentsMaxNum = new Integer(request.getParameter("comments-num"));
+    } catch (NumberFormatException e) {
+      System.err.println("Num of comments wasn't well defined.");
+    }
+    
+    comments.clear();
+    for (Entity entity : results.asIterable()) {
```

Comment:
Are you familiar with the [`FetchOptions`](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/FetchOptions.html) construct? 

You can call `asIterable` with `FetchOptions` and specify the limit, instead of checking it yourself by decreasing `commentsMaxNum` in every iteration. 

For example:
`datastoreService.prepare(query).asList(FetchOptions.Builder.withLimit(10));`

Reply:
I wasn't, it's great!
I used it, thanks